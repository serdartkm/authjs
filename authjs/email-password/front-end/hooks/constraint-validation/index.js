import { h } from "preact";

import useUserNameConstraint from "./useUserNameConstraint";
import usePasswordConstraint from "./usePasswordConstraint";
import useEmailConstraint from "./useEmailConstraint";

export default function useConstraintValidation({ email, password, userName }) {
  const { userNameValidation } = useUserNameConstraint({ userName });
  const { passwordValidation } = usePasswordConstraint({ password });
  const { emailValidation } = useEmailConstraint({ email });

  return {
    emailValidation,
    passwordValidation,
    userNameValidation
  };
}
