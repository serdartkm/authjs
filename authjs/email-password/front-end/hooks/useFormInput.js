import {h} from 'preact'
import {useState} from 'preact/hooks'

export default function useFormInput(state){

    const [value,setValue]= useState(state)

    function onInput(e) {
        const { name } = e.target;
   
        setValue((prevState)=>{ return {...prevState, [name]: e.target.value }});
      }

    return {onInput, ...value}

}