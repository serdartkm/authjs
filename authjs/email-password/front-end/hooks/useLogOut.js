import { h } from "preact";
import { useState } from "preact/hooks";
import useToken from "./useToken";

export default function useLogOut() {
  const { removeToken } = useToken();
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);
  const [result, setResult] = useState(null);

  async function logOut({ token, email }) {
    try {
      setLoading(true);
      const response = await fetch("/logout", {
        method: "POST",
        body: JSON.stringify({ token })
      });
      setResult(response.json());
      removeToken(email);
      setLoading(false);
    } catch (e) {
      setLoading(false);
      setError(e);
    }
  }

  return { result, loading, logOut, error };
}
