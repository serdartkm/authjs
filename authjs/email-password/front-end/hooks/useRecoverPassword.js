import { h } from "preact";
import { useState } from "preact/hooks";

export default function useRecoverPassword() {
  const [loading, setLoading] = useState(false);
  const [result, setResult] = useState(false);
  const [error, setError] = useState(null);

  async function recoverPassword({ email }) {
    try {
      setLoading(true);
      const response = await fetch("/recover", {
        method: "POST",
        body: JSON.stringify({ email })
      });
      setResult(response.json());
      setLoading(false);
    } catch (e) {
      setLoading(false);
      setError(e);
    }
  }
  return { recoverPassword, result, error, loading };
}
