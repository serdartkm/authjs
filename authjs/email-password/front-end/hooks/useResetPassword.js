import { h } from "preact";
import { useState } from "preact/hooks";
import decode from "jwt-decode";
import useToken from "./useToken";

export default function useResetPassword() {
  const { saveToken } = useToken();
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);
  async function resetPassword({ password, token }) {
    try {
      setLoading(true);
      const response = await fetch("/change", {
        method: "POST",
        body: JSON.stringify({ password, token })
      });
      const decoded = decode(response.json().token);
      saveToken(decoded.email, response.json().token);
      setLoading(false);
    } catch (e) {
      setLoading(false);
      setError(e);
    }
  }
  return { resetPassword, loading, error };
}
