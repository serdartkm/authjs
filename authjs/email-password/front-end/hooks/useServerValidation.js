/* eslint-disable import/no-unresolved */
import {h} from 'preact'
import {useState} from 'preact/hooks'
import { signupResponse } from "http-response-status";


export default function useServerValidation (){

    const [serverValidation, setServerValidation]= useState({
        email: { valid: true, message: "" },
        userName: { valid: true, message: "" }
      })


   function validate (response){
    if (response !== null && response !== undefined) {
        const {message}  =response.json()
        
                    if (response.status === signupResponse.BadRequest.status) {
                      if (response.code === signupResponse.BadRequest.USERNAME_TAKEN.code) {
                        setServerValidation(prevState => {
                          return {
                            ...prevState,
                            userName: { valid: false, message}
                          };
                        });
                      } else if (
                        response.code === signupResponse.BadRequest.EMAIL_TAKEN.code
                      ) {
                        setServerValidation(prevState => {
                          return {
                            ...prevState,
                            email: { valid: false, message }
                          };
                        });
                      } else if (
                        response.code === signupResponse.BadRequest.EMAIL_USERNAME_TAKEN.code
                      ) {
                        setServerValidation({
                          email: { valid: false, message },
                          userName: { valid: false, message }
                        });
                  
                      }
                    }
                  }
        
   }
    return {serverValidation,validate}

}
