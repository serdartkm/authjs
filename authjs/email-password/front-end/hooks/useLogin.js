import { h } from "preact";
import { useState } from "preact/hooks";
import useToken from "./useToken";
import useFormInput from './useFormInput'

export default function useLogin() {
  const { saveToken } = useToken;
  const [loading, setLoading] = useState(false);
  const [emailValidation,setEmailValidation] =useState({valid:true,message:''})
  const [passwordValidation,setPasswordValidation]=useState({valid:true,message:''})
  const [error, setError] = useState(null);
  const {value, onInput} = useFormInput({ email: "", password: "" });

  async function login({ password, email }) {
    if (value.email === "" && value.password !=='') {
      setEmailValidation({ valid: false, message: "email or username not provided" });
  
    }
    else
    if (value.password === "" && value.email !=='') {
      setPasswordValidation({ valid: false, message: "password not provided" });
    
    }
    else
    if(value.email==='' && value.password===''){
      setEmailValidation({valid:false, message:'email or username not provided'})
      setPasswordValidation({ valid: false, message: "password not provided" });
   
    }
  
    else{

  
    
    try {
      setLoading(true);
      const response = await fetch("/login", {
        method: "POST",
        body: JSON.stringify({ password, email })
      });
  
      if (response.ok) {
    
        saveToken(email, response.json().token);
        setLoading(false);
        return  
      }
      // validation error
      if (response.status === 401) {
        setEmailValidation({valid:false, message:'email,username or password did not match'})
        setPasswordValidation({ valid: false, message: 'email,username or password did not match' });
      }
      setLoading(false);
    } catch (e) {
      setError(e);
      setLoading(false);
    }
  }
  }

  return {
    loading,
    error,
    login,
    onInput,
  ...value,
    passwordValidation,
    emailValidation
  };
}
