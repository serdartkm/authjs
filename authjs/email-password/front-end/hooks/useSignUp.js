/* eslint-disable import/no-unresolved */
import { h } from "preact";
import { useState, useEffect, useContext } from "preact/hooks";
import { SnackMessageContext } from "components/app-provider";
import useToken from "./useToken";
import useConstraintValidation from "./constraint-validation";
import useFormInput from "./useFormInput";
import useServerValidation from "./useServerValidation";

export default function useSignUp() {
  const snackMessageContext = useContext(SnackMessageContext);

  const { userName, password, email, onInput } = useFormInput({
    email: "",
    password: "",
    userName: ""
  });
  const {
    emailValidation,
    passwordValidation,
    userNameValidation
  } = useConstraintValidation({ userName, email, password });

  const { saveToken } = useToken();
  const [loading, setLoading] = useState(false);
  const [response, setResponse] = useState(null);
  const { serverValidation, validate } = useServerValidation(response);

  function signUp() {
    async function callapi() {
      if (
        emailValidation.valid &&
        userNameValidation.valid &&
        passwordValidation.valid &&
        email !== "" &&
        userName !== "" &&
        password !== ""
      ) {
        try {
          setLoading(true);

          const res = await fetch("/signup", {
            method: "POST",
            body: JSON.stringify({ email, userName, password })
          });
          await setResponse(res);

          setLoading(false);
        } catch (e) {
          const err = e;

          snackMessageContext.setMessage(err.message);
        }
      }
    }
    callapi();
  }

  useEffect(() => {
    validate(response);
    if (response !== null && response !== undefined && response.status.ok) {
      saveToken(response.json().token);
    }
  }, [response]);

  return {
    signUp,
    onInput,
    loading,

    password,
    email,
    userName,
    emailValidation,
    passwordValidation,
    userNameValidation,
    serverValidation
  };
}
