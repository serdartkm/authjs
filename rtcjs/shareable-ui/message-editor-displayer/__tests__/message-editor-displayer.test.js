

import { h } from 'preact'
import {useState} from 'preact/hooks'
import { render, fireEvent,cleanup } from '@testing-library/preact'
import renderer from 'preact-render-to-string'
import MessageEditorDisplayer from '../index'

import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)


describe('MessageEditorDisplayer component', () => {

    it("Should render correctly", () => {
        const tree = renderer(MessageEditorDisplayer)
        expect(tree).toMatchSnapshot();
    })
    it('on user interaction message-editor behaves correctly',()=>{
        const MessageController =()=>{
          const [message,setMessage]=useState('')
          const sendMessage=()=>{
            setMessage('')
          }
          return(
            <MessageEditorDisplayer id={0} message={message} onMessageChange={(e)=>setMessage(e.target.value)} sendMessage={sendMessage} />
          )
        }
        const {getByPlaceholderText,getByTestId } = render( <MessageController  />,{})
        const messageTextInput =getByPlaceholderText(/Enter message text/i)
        const sendMessageButton =getByTestId('sendMessage0')
        expect(sendMessageButton).toBeDisabled()
        expect(messageTextInput.value).toBe('')
        fireEvent.input(messageTextInput,{ target: { value: 'hello' } })
        expect(messageTextInput.value).toBe('hello')
        expect(sendMessageButton).toBeEnabled()
        fireEvent.click(sendMessageButton)
        expect(messageTextInput.value).toBe('')
        expect(sendMessageButton).toBeDisabled()
    })
})
