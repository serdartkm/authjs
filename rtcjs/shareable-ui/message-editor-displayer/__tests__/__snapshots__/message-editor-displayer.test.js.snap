// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageEditorDisplayer component Should render correctly 1`] = `
"function MessageEditorDisplayer(_ref) {
  var onMessageChange = _ref.onMessageChange,
      message = _ref.message,
      sendMessage = _ref.sendMessage,
      _ref$id = _ref.id,
      id = _ref$id === void 0 ? 0 : _ref$id,
      disabled = _ref.disabled;
  return (0, _preact.h)(&quot;div&quot;, {
    style: {
      display: 'flex'
    }
  }, (0, _preact.h)(&quot;input&quot;, {
    style: {
      flex: 1
    },
    &quot;data-testid&quot;: &quot;message&quot;.concat(id),
    onInput: onMessageChange,
    value: message,
    name: &quot;message&quot;,
    type: &quot;text&quot;,
    placeholder: &quot;Enter message text&quot;
  }), (0, _preact.h)(&quot;div&quot;, {
    style: {
      display: 'flex'
    }
  }, (0, _preact.h)(&quot;button&quot;, {
    &quot;data-testid&quot;: &quot;sendMessage&quot;.concat(id),
    disabled: message === '' || disabled,
    style: {
      marginLeft: 2,
      width: '100%'
    },
    onClick: sendMessage
  }, (0, _preact.h)(&quot;svg&quot;, {
    xmlns: &quot;http://www.w3.org/2000/svg&quot;,
    width: &quot;24&quot;,
    height: &quot;24&quot;,
    viewBox: &quot;0 0 24 24&quot;
  }, (0, _preact.h)(&quot;path&quot;, {
    d: &quot;M2.01 21L23 12 2.01 3 2 10l15 2-15 2z&quot;
  }), (0, _preact.h)(&quot;path&quot;, {
    d: &quot;M0 0h24v24H0z&quot;,
    fill: &quot;none&quot;
  })))));
}"
`;
